docs/data/data-grid/aggregation/aggregation.md

```diff
@@ -4,15 +4,277 @@ title: Data Grid - Aggregation
+<p class="description">Use aggregation functions to aggregate your row values.</p>
 
-> ⚠️ This feature isn't implemented yet. It's coming.
->
-> 👍 Upvote [issue #213](https://github.com/mui/mui-x/issues/213) if you want to see it land faster.
+The aggregation can be modified through the grid interface by opening the column menu and selecting an item on the _Aggregation_ select.
 
-When grouping, you will be able to apply an aggregation function to populate the group row with values.
+The aggregated values will be rendered in a footer row at the bottom of the grid.
+
+:::warning
+⚠️ The footer row will be pinned at the bottom of the grid once [#1251](https://github.com/mui/mui-x/issues/1251) is ready.
+:::
+
+{{"demo": "AggregationBasic.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Pass aggregation to the grid
+
+### Structure of the model
+
+The aggregation model is a lookup where the keys are the field of the columns to aggregate and the values are aggregation items describing how those columns should be aggregated.
+
+An aggregation item is an object where `item.footer` contains the name of the aggregation function to apply on the footers
+and `item.inline` contains the name of the aggregation function to apply on the grouping rows.
+
+:::info
+If you only want to aggregate on the footer, you can use the short notation
+
+```ts
+const model: GridAggregationModel = { gross: { footer: 'sum' } };
+// can be replaced by
+const model: GridAggregationModel = { gross: 'sum' };
+```
+
+:::
+
+### Initialize the aggregation
+
+To initialize the aggregation without controlling it, provide the model to the `initialState` prop.
+
+{{"demo": "AggregationInitialState.js", "bg": "inline"}}
+
+### Controlled aggregation
+
+Use the `filterModel` prop to control the aggregation passed to the grid.
+
+You can use the `onFilterModelChange` prop to listen to changes to the filters and update the prop accordingly.
+
+{{"demo": "AggregationControlled.js", "bg": "inline"}}
+
+## Disable the aggregation
+
+### For all columns
+
+You can disable the aggregation by setting the `disableAggregation` prop to true.
+
+It will disable all features related to the aggregation, even if a model is provided.
+
+{{"demo": "AggregationDisabled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### For some columns
+
+In case you need to disable the aggregation on specific column(s), set the `aggregatable` property on the respective column definition (`GridColDef`) to `false`.
+In the example below, the `title` and `year` columns can not be aggregated.
+
+{{"demo": "AggregationColDefAggregable.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Aggregation label
+
+### Choose the column for the label
+
+Use the `aggregationFooterLabelField` to choose on which column the label of the aggregation should be rendered.
+
+{{"demo": "AggregationLabelColumn.js", "bg": "inline"}}
+
+### Customize the content of the label
+
+TODO
```

Comment:
Same as above
If we do decide to keep the label on a single column, then the prop `aggregationFooterLabelField` allows to decide which column it will be (`aggregationFooterLabelField="title"` for instance)

And to customize the content of the label, you have either:

- the label property of the aggregation function 
- the `aggregationMultiFunctionLabel` locale key 
- the `aggregationFooterLabel` whgich is a function taking the current functions aggregated on this row and return the label

I will probably remove the 3rd one and make the 2nd a function to only have 2 apis.

___

But same as above, let's first decide where we put the label